.github/workflows/lint.yml

```
@@ -0,0 +1,42 @@
+name: Lint
+
+on:
+  push:
+    paths:
+      - '*.py'
+  pull_request:
+    branches: ["main"]
+    paths:
+      - '*.py'
+
+jobs:
+  mypy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.11.2
+          architecture: x64
+      - name: Checkout
+        uses: actions/checkout@v1
+      - name: Install mypy
+        run: pip install mypy
+      - name: Run mypy
+        uses: sasanquaneuf/mypy-github-action@releases/v1
+        with:
+          checkName: 'mypy'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

that seems way overkill?

```suggestion
        run: mypy .
```

---

The advantage of the action is that it will show you fancy line and error information when something goes wrong. It's the same framework as seen in https://github.com/kleinesfilmroellchen/spcasm/actions/runs/4566215832/jobs/8058427309, but for mypy.

---

alright, let's try it :)